### PR TITLE
M3-5772: Cleanup Button story

### PR DESCRIPTION
**Pull request by tiffwong**
## Description

Tasks:

- [x] Remove examples of "Cancel" and "Remove" buttons (also remove related code from the actual component)
- [x] Remove "Loading with Text" story
- [x] Remove references to "compact" and replace with the code for "superCompact"
- [x] Add example of a compact button for secondary buttons only
- [x] Document use cases for the different variants
- [x] Add details to `ArgsTable`

Future tasks to be addressed in a separate PR:

- Figure out why the Component API Button's generated code isn't accurate

## How to test

- Check the buttons throughout the app
- Run `yarn storybook` to see the documentation for Buttons (it's nested inside the UI Elements folder)

### CODE REVIEW

File: packages/manager/src/features/Support/AttachFileListItem.tsx [hunk 1]
```diff
@@ -76,54 +72,52 @@ export const AttachFileListItem: React.FC<CombinedProps> = (props) => {
     file.errors && file.errors.length ? file.errors[0].reason : undefined;
 
   return (
-    <React.Fragment>
```
Comment (tiffwong) [inline packages/manager/src/features/Support/AttachFileListItem.tsx:79]: Remove unnecessary `<React.Fragment>`

**Review by DevDW — APPROVED** [review]
Looks good! The Button docs in Storybook :plusplus:

File: packages/manager/src/components/Button/button.stories.mdx [hunk 2]
```diff
@@ -1,16 +1,162 @@
...
+<!-- TODO: Look into why the code generated from binding the component is not accurate -->
+
+export const Template = (args) => (
+  <Button buttonType="primary" {...args}>
+    I'm a button, style me using the table below
+  </Button>
+);
+
+<Canvas>
+  <Story name="Default">{Template.bind({})}</Story>
```
Comment (HanaXu) [inline packages/manager/src/components/Button/button.stories.mdx:132]: If we set default values for the args here, the extra `Set ...` click goes away:

```
<Story
  name="Default" 
  args={{
    compact: false,
    disabled: false,
    disableRipple: false,
    fullWidth: false,
    loading: false,
    tooltipText: '',
  }}
>
  {Template.bind({})}
</Story>
```

**Comment by HanaXu** [comment]
Can we address these console warnings and errors?

image

image

This one's probably a bit out of scope and can probably be looked into later, but I don't think Storybook should be trying to make calls to the API 🤔 : 
image

**Review by HanaXu — APPROVED** [review]
Looks great! 😎

**Pull request merged**